### ISSUE TITLE
[New implementation] Decide on directory structure

### ISSUE THREAD

**Issue by ivanyu** [aiven/tiered-storage-for-apache-kafka#220]
We currently have `<topic>-<topic_id>/<partition>`. Should it be `<topic_id>-<partition>-<topic>` (like in `org.apache.kafka.server.log.remote.storage.LocalTieredStorage`) or something else?

**Comment by jeqo** [comment]
Looking at https://github.com/aiven/tiered-storage-for-apache-kafka/issues/124, we should consider using `segmentid` as part of the prefix.
The main downside I see is that we lose the hierarchical structure that could be valuable when troubleshooting.

I wonder if we should consider 2 directory structures. One for metadata manifest, and another for log segment and index files. This way we could have

```
bucket/
  prefix/
    topic-topicId/
      partition/
        segmentId.manifest
    segmentId/
      topic-partition-segment.log
      topic-partition-time.index
```

This way we can have use the manifest structure as an index for the segmentid-based structure. wdyt?

**Comment by ivanyu** [comment]
For the record, I'm pretty happy with our current structure and created this ticket to formally have this discussion :)

**Comment by ivanyu** [comment]
> This way we can have use the manifest structure as an index for the segmentid-based structure. wdyt?

Could you please elaborate? Also what we get with this?

**Comment by ivanyu** [comment]
I don't think this the prefixing is relevant for S3 any more (see my comment in https://github.com/aiven/tiered-storage-for-apache-kafka/issues/124#issuecomment-1571995894), but may be still relevant for GCS and Azure, it's unclear and poorly documented.
We'll need to look deeper into this.
